tests/serializer/tests_Serialize.cpp

```diff
@@ -0,0 +1,183 @@
+#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
+#include <doctest/doctest.h>
+#include <rdf4cpp/rdf.hpp>
+
+#include <filesystem>
+#include <format>
+#include <random>
+
+using namespace rdf4cpp::rdf;
+
+TEST_SUITE("Serialize") {
+    template<typename NodeT>
+    void run_ser_test(NodeT node, std::optional<std::string> expected = std::nullopt) {
+        if (!expected.has_value()) {
+            expected = std::string{node};
```

Comment:
I checked for the types how `::operator std::string()` is implemented:
- IRI backend_view writes to preallocated string 
- BNode backend_view writes to preallocated string 
- Literal uses serialize()
- Variable backend_view writes to preallocated string 

Literal is tested with custom input. For the others, we are sure that the write function is correct, I think. 

It also makes the test sensitive to any of them being changed. So, I would say it is good now. :)